microsetta_interface/implementation.py

```diff
@@ -1714,20 +2157,10 @@ def post_claim_samples(*, account_id=None, source_id=None, body=None):
             if sample_survey_output is not None:
                 return sample_survey_output
 
-    # Test if biospecimen consent is required! If Required,
-    # route user to biospecimen consent
-    has_error, consent_required, _ = ApiRequest.get(
-        '/accounts/%s/source/%s/consent/%s' %
-        (account_id, source_id, 'biospecimen'))
-
-    if has_error:
-        return consent_required
-
-    if consent_required["result"]:
-        session["source_id"] = source_id
-        return render_consent_page(account_id, "Biospecimen")
-
-    return _refresh_state_and_route_to_sink(account_id, source_id)
```

Comment:
The state machine was built around a linear progression that doesn't have a notion of multiple types of consents. If you'd prefer to modify the state machine to better reflect the new process, we can.

Reply:
Originally yes but it was subsequently revised to allow for more complex progress. For reference, the prerequisite checking was added in [here](https://github.com/biocore/microsetta-private-api/pull/233) with some discussion [here](https://github.com/biocore/microsetta-private-api/issues/199), though digging through the history, I'm not a concrete description in English of the changes made. 

I think the `redirect` does make sense here so the user remains on the same tab, and this isn't as far as I can tell operating outside of the state machine. But, what I do think is worth considering is revising what the state requirements are for accessing these functions. Right now, they use `SOURCE_PREREQS_MET`, but that isn't accurate; rather they require additional state (the biospecimen consent) which is beyond that of a source by itself. So I do wonder if adding that into the state machine, and adjusting the prereqs here, would the right thing to do so that the machine reflects the hard requirement of this method.